File: tests/align/alloc/align_alloc.cpp
#include <stdlib.h>

#include "../../../include/optitrust.h"

typedef int T;

int main() {
  const int N = 10;
  int *p = (int *) MALLOC1(N, sizeof(int));
  T* q;
  q = (T*) MALLOC3 (N, N, N, sizeof(T));

  return 0;
}

File: tests/align/alloc/align_alloc.ml
open Optitrust
open Prelude


let _ = Run.script_cpp (fun () ->

  !! Align.alloc (lit "64") [cFun "MALLOC1"];
  !! Align.alloc (lit "64") [cFun "MALLOC3"];
  !!! (); (* TODO: Find how to eliminate this reparse *)

)
